File: domain/src/main/java/com/github/gmetal/domain/interactor/UseCase.java
package com.github.gmetal.domain.interactor;

import com.github.gmetal.lib.Notifiable;

public abstract class UseCase<Params, SuccessDomainObject, FailureObject> {

    public abstract void buildUseCase(final Params params, final Notifiable<SuccessDomainObject, FailureObject> notifiable);
}

File: domain/src/main/java/com/github/gmetal/domain/interactor/UseCase.kt
package com.github.gmetal.domain.interactor

import com.github.gmetal.lib.Notifiable

abstract class UseCase<Params, SuccessDomainObject, FailureObject> {

    abstract fun buildUseCase(params: Params, notifiable: Notifiable<SuccessDomainObject, FailureObject>)
}
